### PR TITLE
Update deps and licenses

### PR DESCRIPTION
**Pull request by rockdaboot**
Small cleanups
- update dependencies
- update licenses
- remove obsolete cebpf.DefaultVerifierLogSize (otherwise the linter fails)
- remove --bpf-log-size (the provided value is no longer used)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,13 +1,13 @@
 module github.com/open-telemetry/opentelemetry-ebpf-profiler
 
-go 1.22.2
+go 1.23.1
```
Comment (dmathieu) [inline go.mod:3]: Should we support more than just the latest Go version?
The Go SDK uses the same support policy as for the Go language: https://go.dev/doc/devel/release

Which means supporting 1.22 and 1.23.

Comment (rockdaboot) [reply go.mod:3]: :+1: Makes sense given that the code is used as library by different projects.

Comment (rockdaboot) [reply go.mod]: Reverted the Go version bump and updated the PR title and description.

**Review by athre0z — APPROVED** [review]

**Review by florianl — APPROVED** [review]

**Review by fabled — APPROVED** [review]
LGTM

**Pull request merged**